### PR TITLE
Feature/sup 73 update waybill

### PR DESCRIPTION
**Pull request by IrynaZazyba**
added modal window for update waybill, fixed access to waybill page

### CODE REVIEW

**Review by alexeyoverchenko — APPROVED** [review]

File: frontend/src/validation/WaybillValidationRules.js [hunk 1]
```diff
@@ -42,7 +42,14 @@ export function checkIfRouteExists(apps) {
         errorsFields.push("route");
     }
     return errorsFields;
+}
 
+export function checkIfRouteCalculated(waypoints) {
+    let errorsFields = [];
+    if (waypoints.length === 0) {
```
Comment (eugenmolchanov) [inline frontend/src/validation/WaybillValidationRules.js]: (!waypoints.length)

File: frontend/src/validation/WaybillValidationRules.js [hunk 2]
```diff
@@ -70,3 +77,38 @@ export function validateDriver(driver) {
...
+
+    if (!waybill.driver) {
+        errorsFields.push("driver");
+    }
+
+    if (!waybill.car) {
+        errorsFields.push("car");
+    }
+
+    if (waybill.applications.length === 0) {
```
Comment (eugenmolchanov) [inline frontend/src/validation/WaybillValidationRules.js]: (!waybill.applications.length)

File: frontend/src/validation/WaybillValidationRules.js [hunk 3]
```diff
@@ -70,3 +77,38 @@ export function validateDriver(driver) {
...
+        errorsFields.push("apps");
+    }
+
+    let notDeleted = JSON.parse(JSON.stringify(waybill.applications)).filter(app => !app.deleteFromWaybill);
+    if (notDeleted.length === 0) {
+        errorsFields.push("apps");
+    }
+
+
+    if (waybill.route.wayPoints.length === 0) {
```
Comment (eugenmolchanov) [inline frontend/src/validation/WaybillValidationRules.js]: (!waybill.route.wayPoints.length)

**Review by eugenmolchanov — APPROVED** [review]

File: src/main/java/by/itech/lab/supplier/service/impl/ApplicationServiceImpl.java [hunk 4]
```diff
@@ -170,7 +171,6 @@ public void setItemInApplicationAcceptedAt(final List<Long> ids) {
     }
 
     @Override
-    @Transactional
```
Comment (eugenmolchanov) [inline src/main/java/by/itech/lab/supplier/service/impl/ApplicationServiceImpl.java:173]: Make sure saveAll is executed within transaction.

**Pull request merged**